pallets/file-system/src/utils.rs

```diff
@@ -740,6 +765,16 @@ where
                 storage_request_metadata.size,
             )?;
 
+            // Get the file metadata to insert into the Provider's trie under the file key.
+            let file_metadata = storage_request_metadata.clone().to_file_metadata();
+            let encoded_trie_value = file_metadata.encode();
+            expect_or_err!(
+                file_keys_and_metadatas.try_push((file_key.0, encoded_trie_value)),
+                "Failed to push file key and metadata",
+                Error::<T>::FileMetadataProcessingQueueFull,
+                result
```

Comment:
Why shouldn't this ever fail?

Reply:
Both Bounded vectors are limited by the same constant (`file_keys_and_proofs` used in the for loop and `file_keys_and_metadatas` where we push the values) so the calls to `try_push` will never actually fill the vector.